sustainml_cpp/src/cpp/orchestrator/OrchestratorNode.cpp

```diff
@@ -113,6 +115,19 @@ OrchestratorNode::OrchestratorNode(
     task_man_(new TaskManager()),
     participant_listener_(new OrchestratorParticipantListener(this))
 {
+    req_res_ = new core::RequestReplier([this](void* input)
+                    {
+                        ResponseTypeImpl* in = static_cast<ResponseTypeImpl*>(input);
+                        types::ResponseType res;
+                        res = in;
+
+                        {
+                            std::lock_guard<std::mutex> lock(this->mtx_);
+                            this->res_ = res;
```

Comment:
It is not so intutitive that the new operator `=` in `RequestType` and `ResponseType` takes a `RequestTypeImpl*` or `ResponseTypeImpl*`. It should take a `RequestTypeImpl` (not a pointer) . Please, change that and this line should be
`this->res_ = *sample_in`

Same suggestion applies for the OrchestratorNode